index.js

```diff
@@ -170,11 +170,18 @@ function parseIso8601Date(string) {
   }
 }
 
-var monthAbreviations = MONTH_NAMES.map(function (name) { return name.substr(0, 3); });
 exports.monthFromName = monthFromName;
 function monthFromName(month) {
-  var monthIndex = month.length === 3 ? monthAbreviations.indexOf(month) : MONTH_NAMES.indexOf(month);
-  return monthIndex >= 0 ? monthIndex : null;
+  var monthIndex;
+  var monthCount = 0;
+  var monthAbreviations = MONTH_NAMES.map(function (name) { return name.substr(0, month.length); });
+  var monthPosition = monthAbreviations.indexOf(month);
+  while ( monthPosition !== -1 && monthCount <= 2 ) { // If more than one found, stop looking.
```

Comment:
You shouldn't need a loop here.  You can just do something like:

```js
var monthIndex = monthAbreviations.indexOf(month);
if (monthIndex !== -1 && monthAbreviations.indexOf(month, monthIndex + 1) === -1) {
  return monthIndex;
}
return null;
```

This should save you from needing to declare variables for monthCount and monthPosition